mono/mini/mini-generic-sharing.c

```diff
@@ -3347,6 +3358,14 @@ mini_type_is_reference (MonoType *type)
 	return mono_type_is_reference (type);
 }
 
+gboolean
+mini_method_needs_mrgctx (MonoMethod *m)
+{
+	if (mono_class_is_ginst (m->klass) && MONO_CLASS_IS_INTERFACE (m->klass) && !(m->flags & METHOD_ATTRIBUTE_ABSTRACT))
```

Comment:
we should probably introduce a helper function `mono_method_is_default_method` (this can be done in a later PR)